mountpoint-s3/src/cli.rs

```diff
@@ -511,9 +519,40 @@ impl CliArgs {
         filesystem_config.cache_config = self.cache_config();
         filesystem_config.mem_limit = self.mem_limit();
         filesystem_config.use_upload_checksums = self.should_use_upload_checksum(s3_personality);
+
+        if let Some(username) = &self.run_as_user {
+            match Self::resolve_user_group(username) {
+                Ok((uid, gid)) => {
+                    filesystem_config.uid = uid.as_raw();
+                    filesystem_config.gid = gid.as_raw();
+                    tracing::info!("Running Mountpoint as user: {}", username);
+
+                    // Handle errors
+                    if let Err(e) = setgid(gid) {
+                        tracing::error!(error=?e, "Failed to set GID");
+                    }
+                    if let Err(e) = setuid(uid) {
+                        tracing::error!(error=?e, "Failed to set UID");
+                    }
```

Comment:
Sorry, I didn't explain well earlier. I think this method should only handle preparing the `S3FilesystemConfig` and should not have side effects on the process (such as setting UID/GID). I think that should happen elsewhere - perhaps very early in Mountpoint's lifetime?

We need to handle errors here. Logging/tracing is not enough, we should exit with failure.